### PR TITLE
Implement code points validator

### LINKED ISSUE #875: maxCodePoints / minCodePoints (UTF-32 code points)
The length limit of VARCHAR in some RDBs is the number of UTF-32 code points.
maxLength counts an emoji and some kanji as two.

Password requirements by NIST:

https://pages.nist.gov/800-63-3/sp800-63b.html

>  Unicode [ISO/ISC 10646] characters SHOULD be accepted as well. To make allowances for likely mistyping, verifiers MAY replace multiple consecutive space characters with a single space character prior to verification, provided that the result is at least 8 characters in length. Truncation of the secret SHALL NOT be performed. For purposes of the above length requirements, each Unicode code point SHALL be counted as a single character.

This requires we should count an emoji (not compounded ones) or other 4-byte chracters as 1 character in a password.

### PR DESCRIPTION
**Pull request by tats-u**
Fixes #875

Implemented based on graphemes validation.

## Summary by CodeRabbit

* **New Features**
  * Added Unicode code point count validators: `codePoints`, `minCodePoints`, `maxCodePoints`, and `notCodePoints`.
  * Each validator supports custom validation messages and properly handles emoji and other multi-code-point strings.
  * Added public exports for the new validators and `_getCodePointCount`.
* **Documentation**
  * Added API documentation and type references for all new validators and their action/issue types.
* **Tests**
  * Added runtime tests and TypeScript declaration tests covering valid/invalid cases, Unicode edge cases, and `_getCodePointCount`.

### CODE REVIEW

**Comment by fabian-hiller** [comment]
Thank you for creating this PR! I will try to review and merge it in the next weeks. 🙏

**Comment by tats-u** [comment]
I might export the implementation of `_getCodePointCoun` to other validators (e.g. Zod) in the future.

**Comment by fabian-hiller** [comment]
I just want to let you know that I am focusing on our v1 release first before reviewing this PR.

**Comment by tats-u** [comment]
That's not good news. I'll wait for v1.1.0.
And I want to get permission to publish _getCodePointCount as my independent package from you.

**Comment by fabian-hiller** [comment]
Yes, feel free to create your own package. No permission required.

**Comment by tats-u** [comment]
@fabian-hiller rebased to main (no additional changes)

**Comment by tats-u** [comment]
Could you allow this PR to run CIs?

**Comment by fabian-hiller** [comment]
Sorry for my late reply. 😐 I updated the branch and triggered CI.

**Comment by yslpn** [comment]
@tats-u Hi, I recently updated the documentation on a related topic https://github.com/open-circle/valibot/pull/1505, and now there's a third option. Could you please add more cross-references between codePoints, length, and graphemes? I'd like each page to mention the alternatives. And of course, so users understand the differences better.

File: library/src/utils/_getCodePointCount/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './_getCodePointCountCount.ts';
```
Comment (yslpn) [inline library/src/utils/_getCodePointCount/index.ts]: duplecates `Count` in `_getCodePointCountCount`

File: library/src/actions/notCodePoints/notCodePoints.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    kind: 'validation',
+    type: 'not_code_points',
+    reference: notCodePoints,
+    async: false,
+    expects: `!${requirement}`,
+    requirement,
+    message,
+    '~run'(dataset, config) {
+      if (dataset.typed) {
+        const count = _getGraphemeCount(dataset.value);
```
Comment (yslpn) [inline library/src/actions/notCodePoints/notCodePoints.ts]: Copy-paste error, there should be `_getCodePointCount` here

File: website/src/routes/api/(actions)/maxCodePoints/index.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+## Parameters
+
+- `requirement` <Property {...properties.requirement} />
+- `message` <Property {...properties.message} />
+
+### Explanation
+
+With `maxCodePoints` you can validate the number of code points in a string. If the input does not match the `requirement`, you can use `message` to customize the error message.
+
+> Hint: This action is suitable for `VARCHAR` in PostgreSQL and MySQL. Also, [NIST SP 800-63B requires us to count the length of a password in code points.](https://pages.nist.gov/800-63-4/sp800-63b.html#passwordver)
```
[the remark this answers is not in the document]
Comment (tats-u) [reply website/src/routes/api/(actions)/maxCodePoints/index.mdx]: No ID or heading specified. Probably hallucination by fetch block by firewall.

**Comment by yslpn** [comment]
Hi @tats-u  Thanks for working on this. Let's continue.

1) The new APIs from the codePoints family haven't been added to `website/src/routes/api/menu.md`.

2) Reread my comment again: https://github.com/open-circle/valibot/pull/888#issuecomment-5011211484 I think this is important to do.

**Review by yslpn — CHANGES_REQUESTED** [review]

**Comment by yslpn** [comment]
@tats-u Hi. Thank you very much. LGTM, I'll send this PR to Fabian for merge.

Please finalize the comments from the AI ​​code review.

UPD:
I will help and make final corrections, you don't need to do anything.

**Review by yslpn — DISMISSED** [review]

**Review by yslpn — DISMISSED** [review]

**Comment by tats-u** [comment]
I had no time to improve it by myself. Thank you for covering for me.

**Comment by yslpn** [comment]
> I had no time to improve it by myself. Thank you for covering for me.

Thanks for your work. Good feature)

**Comment by fabian-hiller** [comment]
🙏
